aten/src/ATen/native/nested/cuda/NestedTensorTransformerFunctions.cu

```diff
@@ -226,5 +226,211 @@ template void remove_padding_transform0213_kernelLauncher<c10::Half>(
     const int* output_sizes,
     int output_dim,
     const int batch_size);
+
+template<typename T>
+__global__
+void add_padding_1(
+    const T* input,
+    T* output,
+    T padding_value,
+    const int* offsets,
+    const int* input_sizes,
+    int input_dim,
+    int output_sizes_1,
+    const int batch_size)
+{
+  const int batch_id  = blockIdx.x;
+  const int grid_id  = blockIdx.y;
+  const int tid = threadIdx.x + grid_id * 256;
+  const int grainsize = 16 * 256;
+  const int batch_input_offset = offsets[batch_id];
+  const int* sizes_i = input_sizes + batch_id * input_dim;
+  const int batch_output_offset = batch_id * output_sizes_1;
+  for (int ii = 0; ii < (output_sizes_1 / grainsize); ii++) {
+    const int i = ii * grainsize + tid;
+    const int output_offset = batch_output_offset + i;
+    if (i < sizes_i[0]) {
+      output[output_offset] = input[batch_input_offset + i];
+    } else {
+      output[output_offset] = padding_value;
+    }
+  }
+  const int i = (output_sizes_1 / grainsize) * grainsize + tid;
+  if (i < output_sizes_1) {
+    const int output_offset = batch_output_offset + i;
+    if (i < sizes_i[0]) {
+      output[output_offset] = input[batch_input_offset + i];
+    } else {
+      output[output_offset] = padding_value;
+    }
+  }
+}
+
+template<typename T>
+__global__
+void add_padding_2(
```

Comment:
I suspect there's a clever way to use templates to deduplicate these, but we can always do that later